File: Mechanica Engine/src/renderers/color/ColorShader.java
package renderers.color;


import renderers.WorldShader;

public class ColorShader extends WorldShader {
	
	private static final String VERTEX_FILE = "color/colorVertex.txt";
	private static final String FRAGMENT_FILE = "color/colorFragment.txt";

	public int color;

	public ColorShader() {
		super(VERTEX_FILE, FRAGMENT_FILE);
	}

	@Override
	protected void bindAttributes() {
		super.bindAttribute(0, "position");
	}

	@Override
	protected void getAllUniformLocations() {
		getMatrixLocations();
		color = super.getUniformLocation("color");
		
	}

	

}

File: lwjgl3-test/src/shaders/color/ColorShader.kt
package shaders.color

import display.Game
import renderer.WorldShader

/**
 * Created by domin on 28/10/2017.
 */
class ColorShader : WorldShader("color/colorVertex.glsl", "color/colorFragment.glsl") {

    var color: Int = 0

    init {
        start()
        loadProjectionMatrix(Game.projectionMatrix.create())
        loadViewMatrix(Game.viewMatrix.create())
        stop()
    }

    override fun getAllUniformLocations() {
        getMatrixLocations()
        color = super.getUniformLocation("color")
    }

    override fun bindAttributes() {
        super.bindAttribute(0, "position")
        super.bindAttribute(1, "textureCoords")

    }
}